lib/rack/mock_request.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+
+require 'uri'
+require 'stringio'
+require 'cgi/cookie'
```

Comment:
CGI::Cookie is only used in MockResponse, not in MockRequest. URI is only used in MockRequest, not in MockResponse. I'm guessing there are other issues.  If you are going to split these up, please split the tests up as well, and ensure each file only requires the parts it needs.  `SEPARATE=1 ruby test/spec_mock_request.rb` and similar for response should help you figure out which parts are needed.

Reply:
Thanks for this constructive feedback, I agree with your assessment and will update the PR.